thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/filter/ActiveDatasetFilter.java

```diff
@@ -0,0 +1,13 @@
+package org.apache.pinot.thirdeye.anomaly.detection.trigger.filter;
+
+import org.apache.pinot.thirdeye.anomaly.detection.trigger.TriggerEvent;
+import org.apache.pinot.thirdeye.anomaly.detection.trigger.utils.DatasetTriggerInfoRepo;
+
+public class ActiveDatasetFilter implements TriggerEventFilter{
+
+  @Override
+  public boolean isPassed(TriggerEvent e) {
+    DatasetTriggerInfoRepo triggerInfoRepo = DatasetTriggerInfoRepo.getInstance();
+    return triggerInfoRepo.isDatasetActive(e.getDatasetName());
```

Comment:
To decouple the two component, we don't necessarily need this filter.
The event listener just  blindly update the dataset's watermark.
I expect the load is low.

Reply:
There might be a case that a event is associated with a dataset that does not exist in our metadata. I would think it better to keep it.